### DISCUSSION TITLE
About codes of the classification of lanthipeptides

### DISCUSSION THREAD

**Discussion by liyuelan** [antismash/antismash#563] [General]
I want to write a simple script to classify the lanthipeptides. I learned from your rules. From the index.html there are five classes named "lanthipeptide-class-i/ii/iii/iv/v", but I can only find the "class I/II/III" and the corresponding .hmm file; where are the rule and .hmm files of class iv and v, I found some key protein.hmm files in \antismash-master\antismash\detection\hmm_detection\data\. But it turned out they weren't be used at the hmmscan process. So can you tell me a little bit more details？
![image](https://user-images.githubusercontent.com/127102466/223104283-dc306c9b-0ffb-432f-8fdf-c713c1ca485c.png)
![image](https://user-images.githubusercontent.com/127102466/223104733-57764ebc-64af-46e9-bc14-35b463385c0b.png)

**Comment by SJShaw** [comment]
The rules to define protoclusters products are in the `hmm_detection` module, with the class IV/V types being defined here: https://github.com/antismash/antismash/blob/master/antismash/detection/hmm_detection/cluster_rules/strict.txt#L192-L209

The older style of determination within the `lanthipeptides` module is a carryover from antiSMASH 4 (and earlier) and is largely ignored.

**Reply by liyuelan** [reply]
Thank you for your response. I indeed tred it according to the rules. I want to only use hmmscan with this rules to classfied lanthipeptides.  But I met a new problem, that is I can't  figure out how to distinguish class iii and class iv (micKC.hmm is the key). I combined all the .hmmfiles in the path \antismash-master\antismash\detection\hmm_detection\data\ as the hmmscan database (containing micKC.hmm). And used BGC0000563 (venezuelin) and BGC0000551 (SapB) as standard for test. LanKC and LanL are too similar to distinguish them correctly.
Just in case, I plan to copy your hmmscan parameters that can let micKC show in BGC0000551_hmmscan_output and micKC show in BGC0000563_hmmscan_output, so traced it when antimash was runing. I found this in antismash hmmscan subprocess:  

“====/home/lyl/conda/bin/hmmscan --cpu 80 --nobias --cut_ga /home/lyl/conda/lib/python3.8/site-packages/antismash/databases/resfam/Resfams.hmm -
====/home/lyl/conda/bin/hmmscan --cpu 80 --nobias -E 1E-16 /home/lyl/conda/lib/python3.8/site-packages/antismash/detection/genefunctions/data/smcogs.hmm -
====/home/lyl/conda/bin/hmmscan --cpu 80 --nobias --cut_tc /home/lyl/conda/lib/python3.8/site-packages/antismash/detection/nrps_pks_domains/data/nrpspksdomains.hmm -
====/home/lyl/conda/bin/hmmscan --cpu 80 --nobias --cut_tc /home/lyl/conda/lib/python3.8/site-packages/antismash/detection/nrps_pks_domains/data/ksdomains.hmm -
====/home/lyl/conda/bin/hmmscan --cpu 80 --nobias -E 0.25 /home/lyl/conda/lib/python3.8/site-packages/antismash/detection/nrps_pks_domains/data/abmotifs.hmm -”

But all the .hmm database don't contain the micKC as a model. so which command is the rignt one exactly that can distinguish class iii with iv (or contain micKC or not)? Please help me with more clues. Thank you in advance.

**Reply by SJShaw** [reply]
The rules for [classes III and IV](https://github.com/antismash/antismash/blob/master/antismash/detection/hmm_detection/cluster_rules/strict.txt#L183-L201) describe the difference between the two.

The mikKC profile hasn't changed in years and will included in the automatically built bgcseeds.hmm that the detection module runs. That it isn't present in your output means you are either not showing all the of the hmmscan calls or have made some kind of modifications to the code.

**Reply by liyuelan** [reply]
Thank you for your response.
